src/exceptions.jl

```diff
@@ -22,3 +61,38 @@ function Base.showerror(io::IO, ex::RayTaskException, bt=nothing; backtrace=true
     showerror(io, ex.captured.ex, ex.captured.processed_bt; backtrace)
     return nothing
 end
+
+"""
+    OutOfMemoryError <: RayError
+
+Indicates that the node is running out of memory and is close to full.
+
+This exception is thrown when the node is low on memory and tasks or actors are being
+evicted to free up memory.
+"""
+struct OutOfMemoryError <: RayError
+    msg::String
+end
+
+function Base.showerror(io::IO, ex::OutOfMemoryError)
+    print(io, "$OutOfMemoryError: $(ex.msg)")
```

Comment:
ah okay, I see: this one has a more generic name so it's better to qualify it.  I'd suggest that we should always just qualify it (so that the text of the message doesn't change depending on whether the name has been brought into scope locally or not...)
```suggestion
    print(io, "Ray.OutOfMemoryError: $(ex.msg)")
```

Reply:
Seems odd that if someone does `using Ray: OutOfMemoryError` that we would still show `Ray.OutOfMemoryError` for the exception.

Reply:
I don't have a strong opinion about this, it just confused me when I read it initially.  I've approved the PR so use your judgement here.

Reply:
I've opted to keep the interpolated types as is for the reasons outlined in https://github.com/beacon-biosignals/Ray.jl/pull/191#discussion_r1353186429.